sample_apps/go/query-common.go

```diff
@@ -26,8 +26,11 @@ func write(f *os.File, s string) {
 	}
 }
 
-func processScalarType(data *timestreamquery.Datum) string {
-	return *data.ScalarValue
+func processScalarType(datum *timestreamquery.Datum) string {
+	if datum.NullValue != nil && *datum.NullValue {
```

Comment:
The * signifies a pointer right ? Why do we need these two checks ?

Reply:
`*datum.NullValue` is a check that `*datum.NullValue` is true (it could be false as well)

Reply:
Gotcha - so it's just a short-circuiting